src/js/bootstrap.js

```diff
@@ -1,63 +1,114 @@
+
+  if (location.host === 'console.redhat.com') {
+    trustarcScript.src = '//static.redhat.com/libs/redhat/marketing/latest/trustarc/trustarc.js';
+  } else {
+    trustarcScript.src = '//static.redhat.com/libs/redhat/marketing/latest/trustarc/trustarc.stage.js';
+  }
 
-// start auth asap
-const libjwt = auth();
+  document.body.appendChild(trustarcScript);
+};
 
-function noop() {}
+const App = () => {
+  const modules = useSelector(({ chrome }) => chrome?.modules);
+  const scalprumConfig = useSelector(({ chrome }) => chrome?.scalprumConfig);
+  const documentTitle = useSelector(({ chrome }) => chrome?.documentTitle);
+  const dispatch = useDispatch();
+  const [jwtState, setJwtState] = useState(false);
 
-//Add redhat font to body
-document.querySelector('body').classList.add('pf-m-redhat-font');
+  useEffect(() => {
+    initializeAccessRequestCookies();
+    const libjwt = libjwtSetup();
+    libjwt.initPromise.then(() => setJwtState(true));
 
-libjwt.initPromise.then(() => {
-  libjwt.jwt
-    .getUserInfo()
-    .then((...data) => {
-      sentry(...data);
-    })
-    .catch(noop);
-});
+    window.insights = createChromeInstance(libjwt, window.insights);
 
-window.insights = window.insights || {};
+    if (typeof _satellite !== 'undefined' && typeof window._satellite.pageBottom === 'function') {
+      window._satellite.pageBottom();
+      registerUrlObserver(window._satellite.pageBottom);
+    }
 
-window.insights = createChromeInstance(libjwt, window.insights);
+    trustarcScriptSetup();
 
-if (typeof _satellite !== 'undefined' && typeof window._satellite.pageBottom === 'function') {
-  window._satellite.pageBottom();
-  registerUrlObserver(window._satellite.pageBottom);
-}
+    axios
+      .get(`${window.location.origin}${isBeta() ? '/beta' : ''}/config/chrome/fed-modules.json?ts=${Date.now()}`, {
+        headers: {
+          'Cache-Control': 'no-cache',
+          Pragma: 'no-cache',
+          Expires: '0',
+        },
+      })
+      .then((response) => {
+        dispatch(loadModuesSchema(response.data));
+      });
+  }, []);
 
-const trustarcScript = document.createElement('script');
-trustarcScript.id = 'trustarc';
+  useEffect(() => {
+    const title = typeof documentTitle === 'string' ? `${documentTitle} | ` : '';
+    document.title = `${title}console.redhat.com`;
+  }, [documentTitle]);
 
-if (location.host === 'console.redhat.com') {
-  trustarcScript.src = '//static.redhat.com/libs/redhat/marketing/latest/trustarc/trustarc.js';
-} else {
-  trustarcScript.src = '//static.redhat.com/libs/redhat/marketing/latest/trustarc/trustarc.stage.js';
-}
+  return modules && scalprumConfig && jwtState ? <RootApp config={scalprumConfig} /> : null;
+};
 
-document.body.appendChild(trustarcScript);
+const { store } = spinUpStore();
+const pageRoot = document.getElementById('chrome-entry');
 
-import renderChrome from './chrome/render-chrome';
-// render root app
-renderChrome();
+ReactDOM.render(
+  <Provider store={store}>
```

Comment:
You can call `spinUpStore` and access the store out of it.
```suggestion
  <Provider store={spinUpStore()?.store}>
```